PluginFramework/src/Formatter/BaseFormatter.cs

```diff
@@ -0,0 +1,113 @@
+﻿using PluginFramework.src.DataContainer;
+using PluginFramework.src.Enums;
+using PluginFramework.src.EventMessages;
+using PluginFramework.src.Interfaces.PluginTypes;
+using System;
+using System.IO;
+
+namespace PluginFramework.src.Formatter
+{
+    /// <summary>
+    /// Base class for formatting
+    /// </summary>
+    public abstract class BaseFormatter : IFormatter
+    {
+        /// <inheritdoc/>
+        public string Extension => extension;
+
+        /// <summary>
+        /// Instance for the extension
+        /// </summary>
+        private readonly string extension;
+
+        /// <inheritdoc/>
+        public PluginInformation Information => information;
+
+        /// <summary>
+        /// Instance of the plugin information
+        /// </summary>
+        private readonly PluginInformation information;
+
+        /// <inheritdoc/>
+        public event EventHandler<BaseEventArgs> StatusChanged;
+
+        public BaseFormatter(string fileExtension, PluginInformation pluginInformation)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Change the visibility of this constructor to 'protected'.](https://app.codacy.com/manual/simonaberle/XmlFormatter/pullRequest?prid=5494671)